### ISSUE TITLE
Fails with new Flutter embedding system for v2 Android Embedding

### ISSUE THREAD

**Issue by MikeMannox** [henriquearthur/flutter_native_splash#35]
With the latest update to Flutter (I am on `1.13.2`) a new v2 of the Android embedding API was introduced. 

Running:
```cli
flutter create --template app --description "Fluttering" --ios-language "swift" --android-language "kotlin" --org "xyz.flutter" app
```
Generates new an Android project with some subtle differences:

`AndroidManiest.xml` with
```xml
     <!-- Don't delete the meta-data below.
             This is used by the Flutter tool to generate GeneratedPluginRegistrant.java -->
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
```

`MainActivity.kt` with no `onCreate` and a new method:
```kotlin
    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
        GeneratedPluginRegistrant.registerWith(flutterEngine);
    }
```

Adding 
```yaml
dev_dependencies:
  flutter_native_splash: 0.1.9

flutter_native_splash:
  image: assets/splash.png
  color: ffcc11
  fill: true
```
and then executing:
```cli
flutter pub pub run flutter_native_splash:create
```
updates the `drawables/*` and `colors.xml`, but fails to with `MainActivity.kt` 
Sometimes you see the splash for a flicker of a second, followed by a black screen before the Flutter content shows

**Comment by edufolly** [comment]
Probably this is the solution: [#54](https://github.com/henriquearthur/flutter_native_splash/issues/54#issuecomment-589237135)

**Comment by dev01mrrocco** [comment]
Any solution to this problem? Some time ago I upgraded to embedding v2 and got problems in the MainActivity.java class. The biggest problem is that Flutter v 1.22 is constantly complaining about updating:

"Your Flutter application is created using an older version of the Android embedding. It's being deprecated in favor of Android embedding v2. Follow the steps at https://flutter.dev/go/android-project-migration"

**Comment by dev01mrrocco** [comment]
Guys, in order to continue using this package, I found a partial and manual solution to this problem.

Once added to the splash screen with this package, some manual settings can be made to correct this problem:

Adjust your MainActivity.java or kotrin similar to these settings:

```
package br.com.domain.name;

import io.flutter.embedding.android.FlutterActivity;

import androidx.annotation.NonNull;

import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugins.GeneratedPluginRegistrant;

import android.os.Build;
import android.view.ViewTreeObserver;
import android.view.WindowManager;

public class MainActivity extends FlutterActivity {

    @Override
    public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
        GeneratedPluginRegistrant.registerWith(flutterEngine);

        int originalStatusBarColor = 0;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            originalStatusBarColor = getWindow().getStatusBarColor();
            getWindow().setStatusBarColor(0xff016D09);
        }

    }

}
```

And in your AndroidManifest.xml, similar to:

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="br.com.domain.name"
    xmlns:tools="http://schemas.android.com/tools">

    <uses-permission android:name="android.permission.INTERNET"/>

    <application
        tools:replace="android:label"
        android:label="Name"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name=".MainActivity"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">

            <meta-data
                android:name="io.flutter.embedding.android.SplashScreenDrawable"
                android:resource="@drawable/launch_background" />

            <!-- Theme to apply as soon as Flutter begins rendering frames -->
            <meta-data
                android:name="io.flutter.embedding.android.NormalTheme"
                android:resource="@style/NormalTheme"
                />

            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>

        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
</manifest>
```
And finally, add the style, similar to:

```
<?xml version="1.0" encoding="utf-8"?>
<resources>
    <style name="LaunchTheme" parent="@android:style/Theme.Black.NoTitleBar">
        <item name="android:windowBackground">@drawable/launch_background</item>
        <item name="android:windowFullscreen">false</item>
    </style>

    <style name="NormalTheme" parent="@android:style/Theme.Black.NoTitleBar">
        <item name="android:windowBackground">@drawable/launch_background</item>
        <item name="android:windowFullscreen">false</item>
    </style>
</resources>
```

That was a solution that worked for me and I hope it helps in correcting this package.

For more information visit: https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration

**Comment by jonbhanson** [comment]
Should be resolved by 48981828e831768c4ad89a35475e8f9f24e0aca5.
